### DISCUSSION TITLE
Shared subscript after TCA 1.11.0

### DISCUSSION THREAD

**Discussion by seanmrich** [pointfreeco/swift-composable-architecture#3177] [Q&A] [answered]
v1.11.0 deprecated one of the dynamic member lookup subscripts in favor of the failable `Shared` initializer. I'm having a little trouble with this change when accessing a subscript of the shared value. Here's some code to demo the issue. I'm trying to get a `Shared<Tree>?` value:
```swift
struct Tree: Identifiable, Sendable {
  var id: UUID = .init()
  var nodes: [Tree] = []
  
  subscript(path pathIds: [Tree.ID]) -> Tree? {
    get { … }
    set { … }
  }
}

let tree = Shared(Tree())
let node = tree[path: […]]
// 👆'subscript(dynamicMember:)' is deprecated: Use 'Shared($value.optional)' to unwrap optional shared values
```

It's possible to use the other dynamic lookup subscript by explicitly giving the type:
```swift
let tree = Shared(Tree())
let node: Shared<Tree?> = tree[path: […]]
let optional = Shared(node)
```

I use this pattern quite a few times in my app so I'd like to avoid this two-step dance. I suppose I could create an extension that bypasses the dynamic member lookup altogether:
```swift
extension Shared where Value == Tree {
  subscript(path pathIds: [Tree.ID]) -> Shared<Tree>? {
    let node: Shared<Tree?> = self[path: pathIds]
    return Shared(node)
  }
}
```

Of course, this means I need to do this for each type and subscript. Is there a more general, ergonomic solution that I'm missing?

**Comment by mbrandonw** [comment, marked as answer]
Hi @seanmrich, this will be fixed with #3170, which we will merge and release in 1.12 soon.

**Reply by seanmrich** [reply]
Release 1.11.2 fixed the issue for me. It's a tad more verbose for my purposes since you have to wrap the subscript accesses in a `Shared` initializer and sometimes the type checker makes me explicitly provide the `Value` type as well. However, I do understand the motivation. Thanks!
